app/portainer/__module.js

```diff
@@ -400,6 +400,17 @@ angular
         },
       };
 
+      var settingsEdgeCompute = {
+        name: 'portainer.settings.edgeCompute',
+        url: '/edge',
+        views: {
+          'content@': {
+            templateUrl: './views/settings/edge-compute/settingsEdgeCompute.html',
```

Comment:
would be better to use a component here (angularjs)

Reply:
i'll use the `createUserAccessToken` as base, but why is a component a better option here?

Reply:
because it's not using `$scope`, because all the view settings are in one place (in the component definition, instead of in the file)

Reply:
keep in mind that this is NOT part of the settings view, it is a separate item on the sidebar menu, same as the Authentication Settings view, which I used for basis 
![image](https://user-images.githubusercontent.com/16310914/148222931-a50cf9f1-a049-41b7-a015-405a5f92ee11.png)

Reply:
imo it wouldn't hurt to have this implemented in the same way for now, and when we have to migrate the entire settings views, it will be the same approach for both views

Reply:
this is not related. I expect the object to look like this:

`const settingsEdgeComput = {
   name: 'portainer.settings.edgeCompute',
   url: '/edge',
   views: {
     'content@': {
        component: 'settingsEgeComputeView'
     }
}
```

Reply:
> imo it wouldn't hurt to have this implemented in the same way for now, and when we have to migrate the entire settings views, it will be the same approach for both views

it will make it easier if we use as a component. old views there's nothing we can do, for new views we should stick to style guide as it's was written with react migration in mind

Reply:
fair enough, I'll change it to a component